lib/streamlit/elements/lib/column_types.py

```diff
@@ -0,0 +1,1019 @@
+        An optional tooltip that gets displayed when hovering over the column label.
+    """
+    return ColumnConfig(
+        label=label, width=width, help=help, type_config=ListColumnConfig(type="list")
+    )
+
+
+@gather_metrics("column_config.DatetimeColumn")
+def DatetimeColumn(
+    label: str | None = None,
+    *,
+    width: ColumnWidth | None = None,
+    help: str | None = None,
+    disabled: bool | None = None,
+    required: bool | None = None,
+    default: datetime.datetime | None = None,
+    format: str | None = None,
+    min_value: datetime.datetime | None = None,
+    max_value: datetime.datetime | None = None,
+    step: int | float | datetime.timedelta | None = None,
+    timezone: str | None = None,
+) -> ColumnConfig:
+    """Configure a datetime column in ``st.dataframe`` or ``st.data_editor``.
+
+    This is the default column type for datetime values. This command needs to be
+    used in the ``column_config`` parameter of ``st.dataframe`` or
+    ``st.data_editor``. When used with ``st.data_editor``, editing will be enabled
+    with a datetime picker widget.
+
+    Parameters
+    ----------
+
+    label: str or None
+        The label shown at the top of the column. If None (default),
+        the column name is used.
+
+    width: "small", "medium", "large", or None
+        The display width of the column. Can be one of “small”, “medium”, or “large”.
+        If None (default), the column will be sized to fit the cell contents.
+
+    help: str or None
+        An optional tooltip that gets displayed when hovering over the column label.
+
+    disabled: bool or None
+        Whether editing should be disabled for this column. Defaults to False.
+
+    required: bool or None
+        Whether edited cells in the column need to have a value. If True, an edited cell
+        can only be submitted if it has a value other than None. Defaults to False.
+
+    default: datetime.datetime or None
+        Specifies the default value in this column when a new row is added by the user.
+
+    format: str or None
+        A momentJS format string controlling how datetimes are displayed. See
+        `momentJS docs <https://momentjs.com/docs/#/displaying/format/>`_ for available
+        formats. If None (default), uses ``YYYY-MM-DD HH:mm:ss``.
+
+    min_value: datetime.datetime or None
+        The minimum datetime that can be entered.
+        If None (default), there will be no minimum.
+
+    max_value: datetime.datetime or None
+        The maximum datetime that can be entered.
+        If None (default), there will be no minimum.
+
+    step: int, float, datetime.timedelta, or None
+        The stepping interval in seconds. If None (default), the step will be 1 second.
+
+    timezone: str or None
+        The timezone of this column. If None (default),
+        the timezone is inferred from the underlying data.
+    """
+
+    def _format_datetime(value: datetime.datetime | None) -> str | None:
```

Comment:
aren't `_format_datetime` and `_format_date` and `_format_time` basically the same except the parameter type is the same. Can't we just merge them? 
Is there a reason it's this way? If so, might need some comment.

Reply:
Also, should they be in this file? It feels like they belong somewhere else but also don't have a strong opinion on that.

Reply:
Changed this a bit. Since it is just calling the `isoformat`, I don't think we even need any kind of function here.